quantfinlib/train_test_split/_base.py

```diff
@@ -0,0 +1,175 @@
+"""Base Purged and Embargoed Cross-Validation Functions."""
+
+from abc import ABC, abstractmethod
+from numbers import Integral
+from typing import Any, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+
+def _validate_purge_embargo_inputs(
+    train_index: np.ndarray,

```

Comment:
not sure if it is an overoptimization, but np.ndarray is repeated so many times here. If there is also a condition on dimentionality, then it maybe worth creating a custom type using https://github.com/ramonhagenaars/nptyping 

from nptyping import NDArray, Float
np1d = NDArray[Shape["*"], Float]
np2d = NDArray[Shape["*, *"], Float]

and then use it across the library.